minio/policy.py

```diff
@@ -528,36 +528,50 @@ def _get_permissions(s, resource, object_resource, matched_resource,
 
 
 # Returns policy of given bucket name, prefix in given statements.
-def get_policy(statements, bucket_name, prefix=''):
```

Comment:
In general, minio-py policy implementation is older than minio-go and minio-java.  The fixes made in minio-go/minio-java are not fixed in minio-py.

Good thing about minio-go is it has complete unit tests for policy and minio-java has a recent fixes and corrected logic.

I see you are trying to implement https://github.com/minio/minio-java/blob/master/api/src/main/java/io/minio/policy/BucketPolicy.java#L443.

Can you refer the java code and implement the functionality in minio-py?